stan/math/prim/scal/fun/init_threadpool_tbb.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef STAN_MATH_PRIM_SCAL_INIT_THREADPOOL_TBB_HPP
+#define STAN_MATH_PRIM_SCAL_INIT_THREADPOOL_TBB_HPP
+
+// TODO(SW): remove STAN_THREADS guards once Intel TBB is fully
+// mandatory
+
+#include <stan/math/prim/scal/fun/get_num_threads.hpp>
+
+#ifdef STAN_THREADS
+#include <tbb/task_scheduler_init.h>
+#endif
+
+namespace stan {
+namespace math {
+
+/**
+ * Initialize the Intel TBB threadpool through the
+ * tbb::task_scheduler_init object. In case an instance of the
+ * tbb::task_scheduler_object is instantiated, then this has no effect,
+ * which is the default behavior of the Intel TBB library.
+ *
+ * The maximal number of threads is read from the environment variable
+ * STAN_NUM_THREADS using internal::get_num_threads. See conventions
```

Comment:
Well that answers my question lol. Let's put these in the same file, or do we use it elsewhere?

Reply:
like I say... we put it into one file once `map_rect` is TBBified.